src/enforcer.js

```diff
@@ -64,71 +64,40 @@ function isHttpUrl_(url) {
  * A map of attribute names to allowed types.
  * @type {!Object<string, !Object<string, !Function>>}
  */
-let SET_ATTRIBUTE_TYPE_MAP = {
-  // TODO(slekies): Add SVG Elements here
-  // TODO(koto): Figure out what to to with <link>
-  'HTMLAnchorElement': {
-    'href': TrustedTypes.TrustedURL,
-  },
-  'HTMLAreaElement': {
-    'href': TrustedTypes.TrustedURL,
-  },
-  'HTMLBaseElement': {
-    'href': TrustedTypes.TrustedURL,
-  },
-  'HTMLButtonElement': {
-    'formaction': TrustedTypes.TrustedURL,
-  },
-  'HTMLSourceElement': {
-    'src': TrustedTypes.TrustedURL,
-  },
-  'HTMLImageElement': {
-    'src': TrustedTypes.TrustedURL,
-    // TODO(slekies): add special handling for srcset
-  },
-  'HTMLTrackElement': {
-    'src': TrustedTypes.TrustedURL,
-  },
-  'HTMLMediaElement': {
-    'src': TrustedTypes.TrustedURL,
-  },
-  'HTMLInputElement': {
-    'src': TrustedTypes.TrustedURL,
-    'formaction': TrustedTypes.TrustedURL,
-  },
-  'HTMLFormElement': {
-    'action': TrustedTypes.TrustedURL,
-  },
-  'HTMLFrameElement': {
-    'src': TrustedTypes.TrustedURL,
-  },
-  'HTMLIFrameElement': {
-    'src': TrustedTypes.TrustedURL,
-    'srcdoc': TrustedTypes.TrustedHTML,
-  },
-  'HTMLLinkElement': {
-    'href': TrustedTypes.TrustedURL,
-  },
-  'HTMLObjectElement': {
-    'data': TrustedTypes.TrustedScriptURL,
-    'codebase': TrustedTypes.TrustedScriptURL,
-  },
-  'HTMLEmbedElement': {
-    'src': TrustedTypes.TrustedScriptURL,
-  },
-  'HTMLScriptElement': {
-    'src': TrustedTypes.TrustedScriptURL,
-    'text': TrustedTypes.TrustedScript,
-  },
-  'HTMLElement': {
-  },
+let SET_ATTRIBUTE_TYPE_MAP = {};
+
+const typeMap = TrustedTypes.getTypeMapping();
+
+const STRING_TO_TYPE = {
+  'TrustedHTML': TrustedTypes.TrustedHTML,
+  'TrustedScript': TrustedTypes.TrustedScript,
+  'TrustedScriptURL': TrustedTypes.TrustedScriptURL,
+  'TrustedURL': TrustedTypes.TrustedURL,
 };
 
-// Add inline event handlers property names.
-for (let name of getOwnPropertyNames(HTMLElement.prototype)) {
-  if (name.slice(0, 2) === 'on') {
-    SET_ATTRIBUTE_TYPE_MAP['HTMLElement'][name] = TrustedTypes.TrustedScript;
+/**
+ * Converts an uppercase tag name to an element constructor function name.
+ * @param {string} tagName
+ * @return {string}
+ */
+function convertTagToConstructor(tagName) {
+  if (tagName == '*') {
+    return 'HTMLElement';
+  }
+  if (['AUDIO', 'VIDEO'].includes(tagName)) {
```

Comment:
It seems like some Object.getPrototypeOf should let you determine where setters are defined.

Reply:
Good idea, will fix.

Reply:
Actually, abandoned the idea - it would increase the complexity of the function installing the set wrapper (with some tiny runtime cost too), and we fully control the element list anyway, so this should not be a large maintenance burden to keep this in sync with DOM API (which we have to anyway). Keeping this as is, will refactor if this turns out to be an issue.

Reply:
This is now implemented as you suggested (the logic is in `wrapSetter_` for properties, and for attributes `setAttribute` checks with the API (`TT.getAttributeType`) based on a tag name.